sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -36,9 +42,18 @@ public HttpPipeline(HttpPipelineTransport transport, HttpPipelinePolicy[]? polic
             all[policies.Length] = new HttpPipelineTransportPolicy(_transport);
             policies.CopyTo(all, 0);
 
+            _transportIndex = policies.Length;
```

Comment:
Do we need _transportIndex given it's always the length of the array?

Reply:
Nope!